opm/simulators/wells/WellStateFullyImplicitBlackoil.hpp

```diff
@@ -131,6 +131,8 @@ namespace Opm
             perf_skin_pressure_.resize(nperf, 0.0);
 
             int connpos = 0;
+            first_perf_index_.resize(nw+1, 0);
```

Comment:
`connpos` related is from the old `Wells` struct. With the new `first_perf_index_`, the usage of `connpos` can probably be removed.  

And also, unless we use `first_perf_index_` to get the number of connection/perforation for a well, we do not need to make its size `nw+1`. But it really depends how you use it.